docs/fsharp/tutorials/asynchronous-and-concurrent-programming/async.md

```diff
@@ -1,213 +1,360 @@
+```
+
+As you can see, the `main` function has quite a few more calls made. Conceptually, it does the following:
+
+1. Transform the command-line arguments into `Async<unit>` computations with `Array.map`
+2. Create an `Async<'T[]>` that schedules and runs the `countFileBytes` computations in parallel when it runs
+3. Create an `Async<unit>` that is a composition of all work to be run, except the result is ultimately ignored
+4. Explicitly run the last computation with `Async.Start`
 
- // you actually have the result from fetchHtmlAsync now!
- printfn "%s" html
- ```
+When this program runs, `countFileBytes` runs in parallel for each command-line argument. Because asynchronous computations execute independently of program flow, there is no order in which they print their information and finish executing.
 
-2.  `Async.Start` will start an async workflow on another thread, and will **not** await its result.
+## Sequencing asynchronous computations
+
+Because `Async<'T>` is a specification of work rather than a computation that is currently running, you can perform more intricate transformations easily. For example, here is an implementation of the Sequence monad that converts a list of `Async<'T>` into an `Async<'T>` that contains an array of results. It uses the previous `countFileBytes` function:
 
 ```fsharp
-open System
-open System.Net
-  
-let uploadDataAsync url data = 
+module Async =
+    let sequence computations =
+        async {
+            let results = ResizeArray() // Where ResizeArray is the F# alias for System.Collections.List<T>
+
+            for computation in computations do
+                let! result = computation
+                results.Add(result)
+
+            return results.ToArray()
+        }
+
+let countFileBytes path =
     async {
-        let uri = Uri(url)
-        use webClient = new WebClient()
-        webClient.UploadStringAsync(uri, data)
+        let! bytes = File.ReadAllBytesAsync(path) |> Async.AwaitTask
+        let fileName = Path.GetFileName(path)
+        printfn "File %s has %d bytes" fileName bytes.Length
     }
 
-let workflow = uploadDataAsync "https://url-to-upload-to.com" "hello, world!"
+[<EntryPoint>]
+let main argv =
+    argv
+    |> Array.map countFileBytes
+    |> Async.sequence
+    |> Async.RunSynchronously
+    |> ignore
+```
+
+This will execute `countFileBytes` for each item in `argv` in the order of the elements of `argv` rather than executing them in parallel. This may be preferable if you wish to preserve the order of computations.
+
+## Common async functions
+
+Because F# asynchronous computations are a _specification_ of work rather than a representation of work that is already executing, they must be explicitly started with a starting function. There are many [Async starting functions](https://msdn.microsoft.com/library/ee370232.aspx) that are helpful in different contexts. The following section describes some of the more common starting functions.
+
+### Async.Start
```

Comment:
Starting here, based on @bartelink's comments, here's my recommendation:

1. Reorder these different methods from most common use to least common. You can then structure the guidance as "use this, unless <some conditions in your scenario>".
1. I like the "when to use it" and "what to watch for" sections. What will help readers is to point to a "default" choice. (see above item). And, for those choices that aren't the "default", when should you use it?
1. You've done a great job above distinguishing multi-threaded, parallel, and asynchronous code. Where the samples and APIs below use parallel and multi-threaded algorithms, it would help to reinforce those distinctions.

This section is the key for readers. (I learned quite a bit from reading it and the comments). Getting this part crystal clear will pay huge dividends. Mistakes in understanding are mine, but I hope the feedback helps polish it.